internal/app.go

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package internal
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/plugin"
+)
+
+type App struct{}
+
+func NewApp() *App {
+	return &App{}
+}
+
+func (*App) Run() error {
+	ctx := context.Background()
+
+	handler := slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelDebug})
+	logger := slog.New(handler)
+	slog.SetDefault(logger)
+
+	slog.Info("starting NGINX Agent")
+
+	processMonitor := plugin.NewProcessMonitor(&plugin.ProcessMonitorParameters{
+		MonitoringFrequency: 1 * time.Minute,
```

Comment:
should we add configuration for reference?

Reply:
Didn't want to do too much in this PR. So I might make a separate PR where we read in the agent config.

Reply:
the configuration should probably get passed to the app then

Reply:
yeah the `app.go` would read the config and pass the relevant info to the plugins